### PR TITLE
EOS-26271: [F-53B]Support for stack trace include/exclude support-bun…

### PR DESCRIPTION
**Pull request by papan-singh**
…dle filter.

Added COLLECT_STACKTRACE_P flag to include/exclude stackTraces. Default is exclude.
Added -t <dest-dir> , -b <bundle-id and -s|--services for Support Bundle Collection.

Signed-off-by: Papan Kumar Singh <papan.kumarsingh@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by nkommuri — APPROVED** [review]

**Comment by papan-singh** [comment]
[m0reportbug.txt](https://github.com/Seagate/cortx-motr/files/7784366/m0reportbug.txt)
[stacktrace_false_m0reportbug.txt](https://github.com/Seagate/cortx-motr/files/7784367/stacktrace_false_m0reportbug.txt)

**Review by madhavemuri — APPROVED** [review]

**Comment by madhavemuri** [comment]
Looks like need to install motr debug rpms and then backtrace needs to be collected, cureent back traces are empty.

**Comment by papan-singh** [comment]
> Looks like need to install motr debug rpms and then backtrace needs to be collected, cureent back traces are empty.

Installed motr debug rpm and verified the symbol are present
[m0reportbug.txt](https://github.com/Seagate/cortx-motr/files/7801396/m0reportbug.txt)

[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 ~]# rpm -qa | grep motr
cortx-motr-2.0.0-300_git409f711f_any.el7.x86_64
cortx-motr-debuginfo-2.0.0-300_git409f711f_any.el7.x86_64

**Comment by madhavemuri** [comment]
retest this please

**Comment by papan-singh** [comment]
Tested the patch in k8s env
1.
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 ~]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler -b SBbw5f7n9i -t /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps false --stacktrace true
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... *WARNING* ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... *WARNING* /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
*WARNING* /var/crash/: no such directory
*WARNING* /var/log/crash/: no such directory
*WARNING* /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/trace... OK (0m50s)
addb-files /etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb... OK
Please attach these files to the bug report:
  m0reportbug-data.tar.gz
  m0reportbug-cores.tar.gz
  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 ~]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/
cortx-data-headless-svc-ssc-vm-rhev4-1910/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1910  motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1910/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 cortx-data-headless-svc-ssc-vm-rhev4-1910]# ls
_m0reportbug-bundler.stderr  m0reportbug-cores.tar.gz  m0reportbug-data.tar.gz  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 cortx-data-headless-svc-ssc-vm-rhev4-1910]# tar -xvf m0reportbug-data.tar.gz
m0reportbug-data/
m0reportbug-data/addb_files/
m0reportbug-data/addb_files/var/
m0reportbug-data/addb_files/var/motr/
m0reportbug-data/addb_files/etc/
m0reportbug-data/addb_files/etc/cortx/
m0reportbug-data/addb_files/etc/cortx/log/
m0reportbug-data/addb_files/etc/cortx/log/motr/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb/m0d_addb__addb.xz
m0reportbug-data/root-misc/
m0reportbug-data/root-misc/etc/
m0reportbug-data/root-misc/etc/motr/
m0reportbug-data/root-misc/etc/motr/confd.xc
m0reportbug-data/root-misc/etc/sysconfig/
m0reportbug-data/root-misc/etc/sysconfig/motr
m0reportbug-data/root-misc/etc/sysconfig/motr-free-space-monitor
m0reportbug-data/systemd-journal.xz
m0reportbug-data/m0reportbug.txt
2.
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 ~]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler -b SBbw5f7n9i -t /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps false --stacktrace true
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... *WARNING* ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... *WARNING* /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
*WARNING* /var/crash/: no such directory
*WARNING* /var/log/crash/: no such directory
*WARNING* /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/trace... OK (0m50s)
addb-files /etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb... OK
Please attach these files to the bug report:
  m0reportbug-data.tar.gz
  m0reportbug-cores.tar.gz
  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 ~]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/
cortx-data-headless-svc-ssc-vm-rhev4-1910/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1910/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1910  motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1910/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 cortx-data-headless-svc-ssc-vm-rhev4-1910]# ls
_m0reportbug-bundler.stderr  m0reportbug-cores.tar.gz  m0reportbug-data.tar.gz  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1910 cortx-data-headless-svc-ssc-vm-rhev4-1910]# tar -xvf m0reportbug-data.tar.gz
m0reportbug-data/
m0reportbug-data/addb_files/
m0reportbug-data/addb_files/var/
m0reportbug-data/addb_files/var/motr/
m0reportbug-data/addb_files/etc/
m0reportbug-data/addb_files/etc/cortx/
m0reportbug-data/addb_files/etc/cortx/log/
m0reportbug-data/addb_files/etc/cortx/log/motr/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb/
m0reportbug-data/addb_files/etc/cortx/log/motr/7d93f20c8e184930a75c55f22b47f379/addb/m0d_addb__addb.xz
m0reportbug-data/root-misc/
m0reportbug-data/root-misc/etc/
m0reportbug-data/root-misc/etc/motr/
m0reportbug-data/root-misc/etc/motr/confd.xc
m0reportbug-data/root-misc/etc/sysconfig/
m0reportbug-data/root-misc/etc/sysconfig/motr
m0reportbug-data/root-misc/etc/sysconfig/motr-free-space-monitor
m0reportbug-data/systemd-journal.xz
m0reportbug-data/m0reportbug.txt
[stacktrace_false_m0reportbug.txt](https://github.com/Seagate/cortx-motr/files/7801875/stacktrace_false_m0reportbug.txt)
[stacktrace_true_m0reportbug.txt](https://github.com/Seagate/cortx-motr/files/7801876/stacktrace_true_m0reportbug.txt)

File: scripts/install/opt/seagate/cortx/motr/libexec/m0reportbug-bundler [hunk 1]
```diff
@@ -37,13 +37,12 @@ that contains m0reportbug archives
 that contain Motr forensic data
 that can be used for reporting and researching Motr bugs.
 
-Positional arguments:
-  dest-dir    Target directory; defaults to '$DEFAULT_DEST_DIR'.
-  bundle-id   Support bundle ID; defaults to the local host name
-              (`hostname`).
 Optional arguments:
+  -t <dest-dir>    Target directory; defaults to '$DEFAULT_DEST_DIR'.
+  -b <bundle-id>   Support bundle ID; defaults to the local host name
```
Comment (gargpalak048) [inline scripts/install/opt/seagate/cortx/motr/libexec/m0reportbug-bundler:42]: -b,-t,-c,-s are required flags, not optional ones.
Please check #L40, why these are kept under optional arguments?

**Review by gargpalak048 — APPROVED** [review]
LGTM

**Pull request merged**